packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -290,9 +290,15 @@ class Auth {
 
     getDnt() {
         const dntCookieVal = this.get(DNT_COOKIE_NAME)
-        // Only '1' or '0' are valid, and invalid values or lack of cookie must be an undefined DNT
+        // Only '1' or '0' are valid, and invalid values, lack of cookie, or value conflict with token must be an undefined DNT
         let dntCookieStatus = undefined
-        if (dntCookieVal !== '1' && dntCookieVal !== '0') {
+        const accessToken = this.getAccessToken()
+        let hasNoConflictWithToken = true
+        if (accessToken) {
+            const {dnt} = this.parseSlasJWT(accessToken)
```

Comment:
Conditional to handle case when the retrieval of the DNT state occurs before a token is stored in the browser